CMakeLists.txt

```diff
@@ -89,6 +89,31 @@ if(APPLE)
 
   # RPATH stuff
   set(CMAKE_MACOSX_RPATH ON)
+  # Determine if we can link against ML Compute
+  set(MLCOMPUTE_FOUND OFF)
+  execute_process(
+    COMMAND bash -c "xcrun --sdk macosx --show-sdk-path"
+    OUTPUT_VARIABLE _macosx_sdk_path
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+  set(_SDK_SEARCH_PATH "${_macosx_sdk_path}/System/Library/Frameworks/")
+  set(_FRAMEWORK_SEARCH_PATH "/System/Library/Frameworks/")
+
+  find_library(_MLCompute_fwrk_path_ NAMES MLCompute REQUIRED PATHS ${_FRAMEWORK_SEARCH_PATH} NO_DEFAULT_PATH)
+  find_library(_MLCompute_sdk_path_ NAMES MLCompute REQUIRED PATHS ${_SDK_SEARCH_PATH} NO_DEFAULT_PATH)
+
+  if (EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/mlc)
+    set(_MLC_FOLDER_EXISTS TRUE)
+  else()
+    set(_MLC_FOLDER_EXISTS FALSE)
```

Comment:
Nit: Wouldn't `set(_MLC_FOLDER_EXISTS YES)` sounds a bit better than TRUE? (To CMake pairs YES<->NO,  ON<->OFF and TRUE<->FALSE are interchangeable)